web/src/message_list_hover.ts

```diff
@@ -46,11 +43,12 @@ export function message_hover($message_row: JQuery): void {
         can_move_message,
     };
     const $edit_content = $message_row.find(".edit_content");
-    $edit_content.html(render_edit_content_button(args));
-
+    $edit_content.removeClass("can_edit_content can_move_message");
     if (args.is_content_editable) {
+        $edit_content.addClass("can_edit_content");
         $edit_content.attr("data-tooltip-template-id", "edit-content-tooltip-template");
     } else if (args.can_move_message) {
+        $edit_content.addClass("can_move_message");
         $edit_content.attr("data-tooltip-template-id", "move-message-tooltip-template");
     }
```

Comment:
Hmm, would it be better to try to avoid changing the classes at all when the outcome is the same as whatever the current state is, if we're trying to optimize this?

Reply:
Yeah, updated the code to do that. Tested that using various settings in org permissions too.